stock_auto_move/models/stock_move.py

```diff
@@ -13,21 +13,54 @@ class StockMove(models.Model):
         help="If this option is selected, the move will be automatically "
         "processed as soon as the products are available.")
 
+    @api.model
+    def _get_auto_moves_by_pickings(self, auto_moves):
+        """ Group moves by picking.
+        @param auto_moves: stock.move data set
+        @return dict dict of moves grouped by pickings
+        {stock.picking(id): stock.move(id1, id2, id3 ...), ...}
+        """
+        auto_moves_by_pickings = dict()
+        for move in auto_moves:
+            if move.picking_id in auto_moves_by_pickings:
+                auto_moves_by_pickings[move.picking_id] |= move
+            else:
+                auto_moves_by_pickings.update({move.picking_id: move})
+        return auto_moves_by_pickings
+
     @api.multi
     def action_assign(self, no_prepare=False):
-        super(StockMove, self).action_assign(no_prepare=no_prepare)
-        # Transfer all pickings which have an auto move assigned
-        moves = self.filtered(lambda m: m.state == 'assigned' and m.auto_move)
-        todo_pickings = moves.mapped('picking_id')
-        # We create packing operations to keep packing if any
-        todo_pickings.do_prepare_partial()
-        moves.action_done()
+        res = super(StockMove, self).action_assign(no_prepare=no_prepare)
+
+        # Use a variable in the context to whether check
+        # the auto moves or not otherwise we will have an
+        # infinite loop
+        if self.env.context.get('no_check_auto_moves'):
```

Comment:
@zakiuu Can you explain the case as there is maybe another solution than using a context parameter?

Reply:
@rousseldenis no need for the context anymore by fixing the second comment